src/tangle/sources/inx.rs

```diff
@@ -1,2 +1,90 @@
 // Copyright 2023 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use futures::{stream::BoxStream, StreamExt, TryStreamExt};
+
+use super::{BlockData, InputSource, MilestoneData};
+use crate::{
+    inx::{Inx, InxError, MarkerMessage},
+    tangle::{ledger_updates::LedgerUpdateStore, milestone_range::MilestoneRange},
+    types::{
+        ledger::{LedgerSpent, MilestoneIndexTimestamp},
+        tangle::MilestoneIndex,
+    },
+};
+
+#[async_trait]
+impl InputSource for Inx {
+    type Error = InxError;
+
+    async fn milestone_stream(
+        &self,
+        range: MilestoneRange,
+    ) -> Result<BoxStream<Result<MilestoneData, Self::Error>>, Self::Error> {
+        let mut inx = self.clone();
+        Ok(Box::pin(
```

Comment:
[nitpick] It's maybe a matter of personal perference, but you can also call `.boxed()` on the stream.

Reply:
`Box::pin` does not require `futures::StreamExt`